staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh

```diff
@@ -30,40 +32,53 @@ function cleanup {
 
+  local PREFIX=${BASE_PKG}/pkg/apis
+  local INPUT_BASE="--input-base ${PREFIX}"
+  local INPUT_APIS=(
+  ${GROUP_NAME}/
+  ${GROUP_NAME}/${VERSION}
+  )
+  local INPUT="--input ${INPUT_APIS[@]}"
+  local CLIENTSET_PATH="--clientset-path ${CLIENT_PKG}/clientset_generated"
+  echo "generating clientset for group ${GROUP_NAME} and version ${VERSION} at ${GOPATH}/${BASE_PATH}${CLIENT_PKG}"
+  ${CLIENTGEN} ${INPUT_BASE} ${INPUT} ${CLIENTSET_PATH} --output-base=${BASE_PATH} --output-base=${GOPATH}/src/${BASE_PATH}
+  ${CLIENTGEN} --clientset-name="clientset" ${INPUT_BASE} --input ${GROUP_NAME}/${VERSION} ${CLIENTSET_PATH} --output-base=${GOPATH}/src/${BASE_PATH}
+  
+  echo "Building lister-gen"
+  listergen="${PWD}/lister-gen"
+  go build -o "${listergen}" k8s.io/kubernetes/cmd/libs/go2idl/lister-gen
 
-echo "Building lister-gen"
-listergen="${PWD}/lister-gen"
-go build -o "${listergen}" ./cmd/libs/go2idl/lister-gen
+  local LISTER_INPUT="--input-dirs ${BASE_PKG}/pkg/apis/wardle --input-dirs ${BASE_PKG}/pkg/apis/${GROUP_NAME}/${VERSION}"
+  local LISTER_PATH="--output-package ${LISTERS_PKG}"
+  local LISTER_OUTPUT="--output-base ${SERVER_BASE}"
+  echo "generating listers for group ${GROUP_NAME} and version ${VERSION} at ${GOPATH}/${BASE_PATH}${LISTERS_PKG}"
+  ${listergen} ${LISTER_INPUT} ${LISTER_PATH} ${LISTER_OUTPUT}
 
-LISTER_INPUT="--input-dirs k8s.io/sample-apiserver/pkg/apis/wardle --input-dirs k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1"
-LISTER_PATH="--output-package k8s.io/sample-apiserver/pkg/client/listers"
-${listergen} ${LISTER_INPUT} ${LISTER_PATH}
+  echo "Building informer-gen"
+  local informergen="${PWD}/informer-gen"
+  go build -o "${informergen}" k8s.io/kubernetes/cmd/libs/go2idl/informer-gen
 
+  echo "generating informers for group ${GROUP_NAME} and version ${VERSION} at ${GOPATH}/${BASE_PATH}${INFORMERS_PKG}"
 
-echo "Building informer-gen"
-informergen="${PWD}/informer-gen"
-go build -o "${informergen}" ./cmd/libs/go2idl/informer-gen
+  ${informergen} \
+    --input-dirs ${BASE_PKG}/pkg/apis/${GROUP_NAME} --input-dirs ${BASE_PKG}/pkg/apis/${GROUP_NAME}/${VERSION} \
+    --versioned-clientset-package ${CLIENT_PKG}/clientset_generated/clientset \
+    --internal-clientset-package ${CLIENT_PKG}/clientset_generated/internalclientset \
+    --listers-package ${LISTERS_PKG} \
+    --output-package ${INFORMERS_PKG} \
+    --output-base ${SERVER_BASE}
+}
 
-${informergen} \
-  --input-dirs k8s.io/sample-apiserver/pkg/apis/wardle --input-dirs k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1 \
-  --versioned-clientset-package k8s.io/sample-apiserver/pkg/client/clientset_generated/clientset \
-  --internal-clientset-package k8s.io/sample-apiserver/pkg/client/clientset_generated/internalclientset \
-  --listers-package k8s.io/sample-apiserver/pkg/client/listers \
-  --output-package k8s.io/sample-apiserver/pkg/client/informers
-  "$@"
+generate_group wardle v1alpha1
```

Comment:
@sttts you are correct. When generating for multiple groups in its current form the script will generate only for the last group passed.
I can take a look at improving this. One question though as I believe @p0lyn0mial is wanting to use the generated clients and the proposed script will successfully generate the clients which it wouldn't before, is there enough value in the current PR to move forward with it and then follow up with improvements to the generate script? Or would you prefer we keep this open?